nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -115,6 +171,32 @@ def _iter_signals_from_spheres(seeds, niimg, radius, allow_overlap,
         yield X[:, row]
 
 
+def _iter_regions_from_spheres(seeds, radius, allow_overlap, mask_img):
+    """Utility function to iterate over spheres. Used in NiftiSpheresMasker()
+    .inverse_transform()
+
+    Parameters
+    ----------
+    seeds: List of triplets of coordinates in native space
+        Seed definitions. List of coordinates of the seeds in the same space
+        as target_affine (either of mask or if mask is None MNI space).
+    radius: float
+        Indicates, in millimeters, the radius for the sphere around the seed.
+    allow_overlap: boolean
+        If False, an error is raised if the maps overlaps (ie at least two
+        maps have a non-zero value for the same voxel).
+    mask_img: Niimg-like object,
+        Mask as a reference space to project the spheres back into brain space.
+    """
+    adjacency = _get_spheres_rows(seeds, mask_img.affine, mask_img.shape, radius,
+                          allow_overlap)
+
+    for row in adjacency.rows:
+        if len(row) == 0:
+            raise ValueError('Sphere around seed #%i is empty' % i)
+        yield row
```

Comment:
I'm not sure if there are cases where this Error will be raised, and I am not sure how big of an issue this is.  (i'm referencing lines 195ff (but somehow cannot comment on then)

The inverse tranform might create sphere's which are possibly completely masked in the last-step. Without any warning what-so-ever, whereas an Error will occur in the classic transform.

I am not sure (yet) how to implement masking during _get_spheres_rows, but I can give it a try - If it's necessary

Reply:
> I am not sure (yet) how to implement masking during _get_spheres_rows, but I can give it a try - If it's necessary

Sorry I don't understand  what you want to implement here. maybe I don't understand what you mean witth 'masking'.

Reply:
In the current implementation, the masker back projeckts spheres into the space of the mask image. But it doesn't take the brain-mask's content into account. So spheres might be outside of the brain mask.
The brain_mask, however, is applied later in the transformation (final step, where I am calling the `input_data.nifti_maps_masker` utils). 

I was wondering whether this might be a  very confusing behavior for the user. The `transform` is checking if `seeds` are inside of `mask_img` (which makes sense, otherwise there'd be no signal), but the `inverse` isn't.
So given the same `mask_img` and `seeds`, you could use the `inverse_transform` to project spheres to the brain (using something like `np.arange` to just visualize the sphere's location), but might be missing certain spheres. The `transform`, however, will fail.

Reply:
OK, thx for the explanations.
Indeed as you say, we should ensure that the behavior of `transform` and `inverse_transform` is consistent. It sounds like you need to factorize some common code between the two methods ?
 Thx again,

Reply:
I don't think it's as straightforward as that. 

Basically, I tried to repurpose the `transform` code for the `inverse_transform` before (you could see that with the misleading doc-strings and redundant for-loops). 

But I got an idea of how to go on. 

(Sorry I forgot to click on comment..., will tend to it soon)